src/types.ts

```diff
@@ -42,197 +43,69 @@ export type ConfigSeedArgs = PartialConfigSeedArgs & {
-
-export const NetworkModeV = t.union([
-  t.literal('n3h'),
-  t.literal('memory'),
-  t.literal('websocket'),
-])
-export type NetworkMode = t.TypeOf<typeof NetworkModeV>
-
-export const RawNetworkConfigV = t.record(t.string, t.any)
-export type RawNetworkConfig = t.TypeOf<typeof RawNetworkConfigV>
-
-export const NetworkConfigV = t.union([
-  NetworkModeV,
-  RawNetworkConfigV,  // TODO: could make this actually match the shape of networking
-])
-export type NetworkConfig = t.TypeOf<typeof NetworkConfigV>
-
-export const RawLoggerConfigV = t.record(t.string, t.any)
-export type RawLoggerConfig = t.TypeOf<typeof RawLoggerConfigV>
-
-export const RawTracingConfigV = t.record(t.string, t.any)
-export type RawTracingConfig = t.TypeOf<typeof RawTracingConfigV>
-
-export const LoggerConfigV = t.union([
-  t.boolean,
-  t.record(t.string, t.any),
-])
-export type LoggerConfig = t.TypeOf<typeof LoggerConfigV>
-
-export const CloudWatchLogsConfigV = t.partial({
-  region: t.string,
-  log_group_name: t.string,
-  log_stream_name: t.string
-})
-export type CloudWatchLogsConfig = t.TypeOf<typeof CloudWatchLogsConfigV>
-
-export const RawCloudWatchLogsConfigV = t.intersection([CloudWatchLogsConfigV, t.type({
-  type: t.literal('cloudwatchlogs')
-})])
-export type RawCloudWatchLogsConfig = t.TypeOf<typeof RawCloudWatchLogsConfigV>
-
-export const LoggerMetricPublisherV = t.literal('logger')
-export type LoggerMetricPublisher = t.TypeOf<typeof LoggerMetricPublisherV>
-
-export const RawLoggerMetricPublisherV = t.type({
-  type: LoggerMetricPublisherV
-})
-export type RawLoggerMetricPublisher = t.TypeOf<typeof RawLoggerMetricPublisherV>
-
-export const MetricPublisherConfigV = t.union([
-  LoggerMetricPublisherV,
-  CloudWatchLogsConfigV,
-])
-export type MetricPublisherConfig = t.TypeOf<typeof MetricPublisherConfigV>
-
-export const RawMetricPublisherConfigV = t.union([RawCloudWatchLogsConfigV, RawLoggerMetricPublisherV])
-export type RawMetricPublisherConfig = t.TypeOf<typeof RawMetricPublisherConfigV>
-
-export type RawSignalsConfig = {
-  trace: boolean,
-  consistency: boolean,
+// export type PlayerConfigs = ObjectS<PlayerConfig>
+export type PlayerConfig = ConfigSeed
+
+
+
+/*
+InstallAgentsHapps
+there will be one agent generated per each in this list
+## example 1
+one agent, one happ, 4 dnas
+[[['dna1', 'dna2', 'dna3', 'dna4']]]
+
+## example 2
+two agents, one happ, two dnas each
+[[['dna1', 'dna2']], [['dna3', 'dna4']]]
+
+## example 3
+sometimes we can write it like this to make it easier to read
+[
+  agent one
+  [
+    happ one, two dnas
+    ['dna1', 'dna2']
+  ]
+]
+*/
+export type InstallAgentsHapps = InstallHapps[]
+export type InstallHapps = InstallHapp[]
+export type InstallHapp = DnaPath[]
+export type DnaPath = string
+
+// the mirror of InstallAgentHapps
+export type InstalledAgentHapps = InstalledHapps[]
+// the mirror of InstallHapps
+export type InstalledHapps = InstalledHapp[]
+
+// the mirror of AgentHapp, but more javascripty
```

Comment:
What's `AgentHapp`?

Reply:
Oops, `InstallHapp` that should be now
```suggestion
// the mirror of InstallHapp, but more javascripty
```